src/go/rdctl/pkg/snapshot/snapshotter_unix.go

```diff
@@ -88,50 +91,61 @@ func NewSnapshotterImpl() Snapshotter {
 	return SnapshotterImpl{}
 }
 
-func (snapshotter SnapshotterImpl) CreateFiles(appPaths paths.Paths, snapshotDir string) error {
+func (snapshotter SnapshotterImpl) CreateFiles(ctx context.Context, appPaths paths.Paths, snapshotDir string) error {
+	taskRunner := runner.NewTaskRunner(ctx)
```

Comment:
Hmm, I'm not sure why we need a `TaskRunner` here; since everything runs in series anyway, it seems like we can just replace things with a `contextIsDone(ctx)` between each file? (Or we can use an `errgroup.Group` and run multiple copies in parallel.)

Presumably a comment in the PR (as opposed to in the code) explaining this in more detail would be enough.

Reply:
As discussed on slack, we will need something like `TaskRunner` for #5555, so we will keep it for now, even though it's not strictly necessary for this PR.